### ISSUE TITLE
Reprinter does not detect if it generate very long lines

### ISSUE THREAD

**Issue by FulgurIgor** [camfort/fortran-src#116]
I have tried to reprint the following code:
```fortran
      program main
      integer, parameter :: &
      long_variable_name_10 = 1, &
      long_variable_name_11 = 1, &
      long_variable_name_12 = 1, &
      long_variable_name_13 = 1, &
      long_variable_name_14 = 1, &
      long_variable_name_15 = 1, &
      long_variable_name_16 = 1, &
      long_variable_name_17 = 1, &
      long_variable_name_18 = 1, &
      long_variable_name_19 = 1, &
      long_variable_name_20 = 1, &
      long_variable_name_21 = 1, &
      long_variable_name_22 = 1, &
      long_variable_name_23 = 1, &
      long_variable_name_24 = 1, &
      long_variable_name_25 = 1, &
      long_variable_name_26 = 1, &
      long_variable_name_27 = 1, &
      long_variable_name_28 = 1, &
      long_variable_name_29 = 1, &
      long_variable_name_30 = 1, &
      long_variable_name_31 = 1, &
      long_variable_name_32 = 1, &
      long_variable_name_33 = 1, &
      long_variable_name_34 = 1, &
      long_variable_name_35 = 1, &
      long_variable_name_36 = 1, &
      long_variable_name_37 = 1, &
      long_variable_name_38 = 1, &
      long_variable_name_39 = 1
      end
```

And it generates the following output, although the maximum length of the line is 132 characters:
```fortran
program main
  integer, parameter :: long_variable_name_10 = 1, long_variable_name_11 = 1, long_variable_name_12 = 1, long_variable_name_13 = 1, long_variable_name_14 = 1, long_variable_name_15 = 1, long_variable_name_16 = 1, long_variable_name_17 = 1, long_variable_name_18 = 1, long_variable_name_19 = 1, long_variable_name_20 = 1, long_variable_name_21 = 1, long_variable_name_22 = 1, long_variable_name_23 = 1, long_variable_name_24 = 1, long_variable_name_25 = 1, long_variable_name_26 = 1, long_variable_name_27 = 1, long_variable_name_28 = 1, long_variable_name_29 = 1, long_variable_name_30 = 1, long_variable_name_31 = 1, long_variable_name_32 = 1, long_variable_name_33 = 1, long_variable_name_34 = 1, long_variable_name_35 = 1, long_variable_name_36 = 1, long_variable_name_37 = 1, long_variable_name_38 = 1, long_variable_name_39 = 1
end program main
```
```

**Comment by dorchard** [comment]
That's interesting. It looks like it may be pretty printing it there. Could you share with us the procedure which you used to reprint this? (have you been working directly with the library or is this via the external tool)? We don't have anything in the pretty printer (I believe) that will deal with character limits here.

**Comment by FulgurIgor** [comment]
I just run the following command:
```
fortran-src --reprint test.f90
```

**Comment by raehik** [comment]
A post-pretty step is provided in https://github.com/camfort/fortran-src/pull/129 that would solve this. Not awfully nicely, mind you, since it does the bad thing of adding continuations in the middle of words (instead of trying to "time it" to place it between words, at the whitespace). If you'd like a hand building and testing it, let me know.

**Comment by raehik** [comment]
Resolved in #129.

    fortran-src --reprint --split-long test.f90

wraps your example to an ugly, but valid, form:

```fortran
program main
  integer, parameter :: long_variable_name_00 = 1, long_variable_name_01&
     & = 1, long_variable_name_02 = 1, long_variable_name_03 = 1, long_v&
     &ariable_name_04 = 1, long_variable_name_05 = 1, long_variable_name&
     &_06 = 1, long_variable_name_07 = 1, long_variable_name_08 = 1, lon&
     &g_variable_name_09 = 1
end program main
```

Note that `--split-long` isn't on by default.
